torch/utils/cpp_extension.py

```diff
@@ -407,6 +407,18 @@ def unix_wrap_ninja_compile(sources,
             # Use absolute path for output_dir so that the object file paths
             # (`objects`) get generated with absolute paths.
             output_dir = os.path.abspath(output_dir)
+
```

Comment:
How difficult is it to add a test to https://github.com/pytorch/pytorch/blob/master/test/test_cpp_extensions_aot.py? I don't think we need to test the library_dir case because that seems difficult to test, but I feel like we should at least test that `include_dir` is working correctly.

Reply:
@zou3519. 
I think we don't need to add this into auto test script. Since this involves user action as well, the build script (setup.py) is created manually by the user who want to build cpp extension. and different people might have different build scripts based on their files name and location. We'd better test it when doing the code change and rely on people reporting issues if any in this case. It is not like pytorch, we control the build script.

Reply:
The thing I'd like to test here is that "relative include_dirs work". We don't have to check that "arbitrary user configurations work".

I think this is possible, because we test cpp extensions over a custom build script: https://github.com/pytorch/pytorch/tree/master/test/cpp_extensions. We should be able to test that "relative include_dirs work" by adding an includes folder that contains some code to that directory and then adding an "include_dirs" option to the setup.py: https://github.com/pytorch/pytorch/blob/ba0851326c87256a819197c133b44f3e617b0608/test/cpp_extensions/setup.py#L19-L29 .

The problem with relying on people to report issues is that there is the risk that we release a version of PyTorch and break some users' code; that's what happened in this case.